src/graph/visitor/PropertyTrackerVisitor.cpp

```diff
@@ -147,41 +189,26 @@ void PropertyTrackerVisitor::visit(AttributeExpression *expr) {
       if (subLeftExpr->kind() == Expression::Kind::kInputProperty) {
         auto *inputPropExpr = static_cast<InputPropertyExpression *>(subLeftExpr);
         auto &edgeAlias = inputPropExpr->prop();
-        propsUsed_.edgePropsMap[edgeAlias][kUnknownEdgeType].emplace(propName);
+        propsUsed_.insertEdgeProp(edgeAlias, kUnknownEdgeType, propName);
       }
+      break;
+    }
+    case Expression::Kind::kFunctionCall: {  // properties(t3).name
+      // TODO(jmq) determine whether it is a vertex or edge
+      break;
     }
     default:
       break;
   }
 }
 
 void PropertyTrackerVisitor::visit(FunctionCallExpression *expr) {
-  static const std::unordered_set<std::string> kVertexIgnoreFuncs = {"id"};
-  static const std::unordered_set<std::string> kEdgeIgnoreFuncs = {
-      "src", "dst", "type", "typeid", "rank"};
+  static const std::unordered_set<std::string> ignoreFuncs = {
+      "src", "dst", "type", "typeid", "id", "rank", "length"};
```

Comment:
That is a bad implementation and bad fix. 
We should not ignore the inner expression behavior. It's really ambiguous.